blockly-core/core/block_space_editor.js

```diff
@@ -283,13 +283,14 @@ Blockly.BlockSpaceEditor.prototype.bumpOrDeleteOutOfBoundsBlocks_ = function() {
   }
 
   var MARGIN = 25;
+  var MARGIN_TOP = 15;
   var overflow;
   var blocks = this.blockSpace.getTopBlocks(false);
   for (var b = 0, block; block = blocks[b]; b++) {
     var blockXY = block.getRelativeToSurfaceXY();
     var blockHW = block.getHeightWidth();
     // Bump any block that's above the top back inside.
-    overflow = metrics.viewTop + MARGIN - blockHW.height -
+    overflow = metrics.viewTop + MARGIN_TOP -
```

Comment:
That's not the assumption here; instead, in the dragged-past-top case I'm bumping based on the block-top, not the block-bottom.  That way when you drag a group of blocks off the top, the whole block group gets bumped back onto the screen, not just the bottom edge of the group.